pkg/sql/schemachanger/scbuild/builder.go

```diff
@@ -119,6 +137,10 @@ func (b *buildContext) build(ctx context.Context, n tree.Statement) (output scpb
 			}
 		}
 	}()
+	// Set up the metadata associated with the current statement,
+	// which will be used to track information about what generated
+	// the current set of elements.
+	b.statementMetaData = *b.addStatementMetaData(n)
```

Comment:
nit: this pattern feels a little awkward. I'd prefer if you either made it a function instead of a method and then update here or if you had the method clearly update the field.